pkg/ddevapp/silverstripe.go

```diff
@@ -0,0 +1,74 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/ddev/ddev/pkg/fileutil"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"os"
+	"path/filepath"
+)
+
+// see https://github.com/ddev/ddev/pull/5058 if the bin folder changes.
+// isSilverstripeApp returns true if the app is of type Silverstripe
+func isSilverstripeApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, app.ComposerRoot, "vendor/bin/sake"))
+}
+
+func silverstripePostStartAction(app *DdevApp) error {
+	// We won't touch env if disable_settings_management: true
+	if app.DisableSettingsManagement {
+		return nil
+	}
+	envFilePath := filepath.Join(app.AppRoot, app.ComposerRoot, ".env")
+	_, envText, err := ReadProjectEnvFile(envFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Unable to read .env file: %v", err)
+	}
+	if os.IsNotExist(err) {
+		err = fileutil.CopyFile(filepath.Join(app.AppRoot, app.ComposerRoot, ".env.example"), envFilePath)
+		if err != nil {
+			util.Debug("Silverstripe: .env.example does not exist yet, not trying to process it")
+			return nil
+		}
+		_, envText, err = ReadProjectEnvFile(envFilePath)
+		if err != nil {
+			return err
+		}
+	}
+	port := "3306"
+	dbConnection := "MySQLDatabase"
+	// Although possible, it is extremely uncommon to use Postgres with Silverstripe.
+	// Thus the option is there to override
+	if app.Database.Type == nodeps.Postgres {
+		dbConnection = "PostgreSQLDatabase"
+		port = "5432"
+	}
+	envMap := map[string]string{
+		"SS_DATABASE_SERVER":        "db",
+		"SS_DATABASE_PORT":          port,
+		"SS_DATABASE_NAME":          "db",
+		"SS_DATABASE_USERNAME":      "db",
+		"SS_DATABASE_PASSWORD":      "db",
+		"SS_ENVIRONMENT_TYPE":       "dev",
+		"SS_DATABASE_CLASS":         dbConnection,
+		"MAILER_DSN":                "smtp://localhost:1025",
+		"SS_DEFAULT_ADMIN_USERNAME": "admin",
+		"SS_DEFAULT_ADMIN_PASSWORD": "password",
+	}
```

Comment:
and would it make sense to add ss_base_url and set it to the DDEV_PROJECT and DDEV_TLD? so that you get `SS_BASE_URL="http://silverstripe.ddev.site"`?

Reply:
That's only for if it can't detect it. Adding it has caused a myriad of problems with HTTPS settings not being applied properly, and if multiple domains are used (Silverstripe Subsites), it creates a lot of "random" issues with the domains added/used.

It's suggested for using CLI commands, but even then, it's not necessary most of the time.

Reply:
good to know. i've only tested it in a single site setup. i was unaware that it might cause some troubles in more complicated setups. then a good choice to leave it out. thank you!